packages/core-progress/core-progress.test.js

```diff
@@ -1,90 +1,106 @@
+  })
 
-test('updates label from indeterminate value', withPage, async (t, page) => {
-  await page.setContent('<core-progress value="Loading..."></core-progress>')
-  t.is(await page.$eval('core-progress', el => el.getAttribute('aria-label')), 'Loading...')
-})
+  it('updates label from value as radial', async () => {
+    await browser.executeScript(() => {
+      document.body.innerHTML = `
+        <core-progress value="0.2" type="radial"></core-progress>
+      `
+    })
+    await expect($('core-progress').getAttribute('aria-label')).toEqual('20%')
+    await browser.executeScript(() => (document.querySelector('core-progress').value = 0))
+    await expect($('core-progress').getAttribute('aria-label')).toEqual('0%')
+    await browser.executeScript(() => (document.querySelector('core-progress').value = 1))
+    await expect($('core-progress').getAttribute('aria-label')).toEqual('100%')
+  })
 
-test('calculates percentage from max', withPage, async (t, page) => {
-  await page.setContent('<core-progress value="0.5"></core-progress>')
-  t.is(await page.$eval('core-progress', el => el.getAttribute('aria-label')), '50%')
-  await page.evaluate(() => (document.querySelector('core-progress').max = 10))
-  t.is(await page.$eval('core-progress', el => el.getAttribute('aria-label')), '5%')
-  await page.evaluate(() => (document.querySelector('core-progress').max = 100))
-  t.is(await page.$eval('core-progress', el => el.getAttribute('aria-label')), '1%')
-})
+  it('updates label from indeterminate value', async () => {
+    await browser.executeScript(() => {
+      document.body.innerHTML = `
+        <core-progress value="Loading..."></core-progress>
+      `
+    })
+    await expect($('core-progress').getAttribute('aria-label')).toEqual('Loading...')
+  })
 
-test('does not trigger change event on max', withPage, async (t, page) => {
-  await page.setContent('<core-progress></core-progress>')
-  await page.evaluate(() => {
-    return new Promise((resolve, reject) => {
-      window.addEventListener('change', reject)
+  it('calculates percentage relative to max', async () => {
+    await browser.executeScript(() => {
+      document.body.innerHTML = `
+        <core-progress value="0.5"></core-progress>
+      `
+    })
+    await expect($('core-progress').getAttribute('aria-label')).toEqual('50%')
+    await browser.executeScript(() => (document.querySelector('core-progress').max = 10))
+    await expect($('core-progress').getAttribute('aria-label')).toEqual('5%')
+    await browser.executeScript(() => (document.querySelector('core-progress').max = 100))
+    await expect($('core-progress').getAttribute('aria-label')).toEqual('1%')
+  })
+
+  it('does not trigger change event on max', async () => {
+    await browser.executeScript(() => {
+      document.body.innerHTML = `
+        <core-progress></core-progress>
+      `
+      document.addEventListener('change', () => (document.body.appendChild(document.createElement('i'))))
```

Comment:
Why are you creating and appending an element to verify if the test passes, instead of just changing a variable `didChange` or something ?

Reply:
this is a good point. the `executeScript(...)` can't access the tests' js scope from the browsers' scope. instead of creating dom elements for assertion, i will try setting a `window.triggered = ...` in the browser and check  `window.triggered` instead. it should be less confusing.

Reply:
Thank you, it's a lot more intuitive now (at least for me :p)!